application/modules/invoices/views/modal_preview_pdf.php

```diff
@@ -0,0 +1,27 @@
+<script type="text/javascript">              
+    $(function () {
+        $('#modal-preview-pdf').modal('show');  
+    });   
+     
+</script>
+
+<div id="modal-preview-pdf" class="modal fade col-xs-12 col-sm-10 col-sm-offset-1 col-md-8 col-md-offset-2"
+     role="dialog" aria-labelledby="modal-preview-pdf" aria-hidden="true">
+    <form class="modal-content">
+        <div class="modal-header">         
+            <a data-dismiss="modal" class="close"><i class="fa fa-close"></i></a>
+            <h3><i class="fa fa-file-pdf-o"></i><?php echo '  ' . trans('invoice') . ': #' . $invoice->invoice_number ; ?></h3>            
+        </div>
+        <div class="modal-body" style="padding: 0; margin: 0; background-color: #ededf0;">
+            <iframe id="iframe_pdf" src="<?php echo site_url('invoices/generate_pdf/' . $invoice_id); ?>#zoom=page-width" width="100%" height="70%" scrolling="no" 
```

Comment:
@Verony-makesIT can you find another solution than an iframe?

Reply:
@nielsdrost7, I assume your question is because of a possible xss security issue with iframes?
I am currently exploring other options such as _embed_ or _object_.
If you think of other or better options for _previewing PDFs_, please let me know...